src/number_format.js

```diff
@@ -595,8 +595,8 @@ class NumberFormat extends React.Component {
       value.length > lastValue.length
       || !value.length ||
       start === end ||
-      (start === 0 && end === lastValue.length) ||
-      (selectionStart === 0 && selectionEnd === lastValue.length)
+      (start <= prefix.length && end >= lastValue.length - suffix.length) ||
```

Comment:
Also the condition `start <= prefix.length && end >= lastValue.length - suffix.length` will return half formatted value when multiple character is used for prefix/ suffix. For example if you used `$$` as prefix.
and the value is `$$3.00`.  And if you replace `$3.00` with character 4. This method will return `$4` which this method is trying to avoid. This might cause some other bug.

Reply:
The point about partially selected prefixes/suffixes is very valid. In my manual tests the overall behaviour in such a case still seemed to be fine, but since I do not know the whole context here I am now suggesting a more defensive approach.
Only adding one additional condition for returning early from `correctInputValue()`, which is if exactly the number without prefix/suffix is selected/changed. If parts of prefix/suffix are part of the selection, this results in no change to the value.